lib/hammer_cli_foreman/virtual_machine.rb

```diff
@@ -0,0 +1,78 @@
+
+module HammerCLIForeman
+
+  class VirtualMachine < HammerCLIForeman::Command
+
+    resource :compute_resources
+    command_name 'virtual_machine'
+    desc _("View and manage compute resource's virtual machines")
+
+
+    class AvailableVirtualMachinesCommand < HammerCLIForeman::ListCommand
+
+      resource :compute_resources, :available_virtual_machines
+      command_name 'list'
+
+      option '--compute-resource-id', 'ID', '',
+             :attribute_name => HammerCLI.option_accessor_name('id'), :required => true
+
+      output do
+        field :name, _("Name")
+        field :id, _("ID")
+      end
+
+      build_options
+    end
+
+    class PowerVmCommand < HammerCLIForeman::Command
+
+      resource :compute_resources, :power_vm
+      command_name 'power'
+
+      option '--compute-resource-id', 'ID', '',
+             :attribute_name => HammerCLI.option_accessor_name('id'), :required => true
+      option "--vm-id", "VM-ID", _("Virtual machine id"), :required => true
```

Comment:
I personally don't like that we've got the same options defined multiple times in this file.

I'd create a command extension for this purpose (in case we'll need to remove them or somehow adjust in the future).

Moreover, I think we need to add `MultiResourceCommand` definition (which is out of the scope of this PR unless you want to do it, so we could actually test it with those commands), because of redundant/wrong automatically generated options such as `--id` and `--name`. Those options are expecting data for a compute resource... resource, but as a user I'd expect that expected data are for virtual machine (`hammer compute-resource virtual-machine info --id VM-ID`).

Reply:
Thanks :) I agree and tried to change my code according to all your comments. I think maybe now it is better and there is no need to add MultiResourceCommand definition. Let me know what do you think about my changes.